### ISSUE TITLE
Test result in the same page as the code, below the exercise description

### ISSUE THREAD

**Issue by faloi** [mumuki/mumuki-platform#207]
I think the current approach is very old school: fill a form, submit and wait for the result to show.

My proposal is to have everything in the same page, and update the results every time the user submits a new version of the code.

**Comment by faloi** [comment]
@flbulgarelli @aguspina @jvillarejo @PalumboN please drop a comment here

**Comment by asanzo** [comment]
I was just talking with Carlos Lombardi regarding this. We're starting a Gobstones course in August, and he doubts whether to use Mumuki or not. Our main concerns are that this time the learners are non-techies, so every UX improvement is one barrier less to go through.
Our two main criticisms are 1) we need either a very straightfoward Register screen or a default guest account, or both. 2) exactly the topic of this issue: At least in the case of a guide, dynamic loading of results. Coding space proportional to expected results, and also inlined . This way the description of the problem can be as big as neccessary, and the user has immediate and intuitive navigation to previous parts of the guide.
In other words, something like this would be great: https://rubymonk.com

**Comment by flbulgarelli** [comment]
@asanzo 

 > we need either a very straightfoward Register screen or a default guest account, or both

We have currently implemented Facebook login in #181 - it is not in production yet but it will be - that allows you to signin with either Github or Faceebok. 

I think that should make login easier, since nearly everyone has a facebook account these days.

**Comment by flbulgarelli** [comment]
On the other hand, yes, I would like to change the current behaviour to display results above the submission and autoscroll. We definetly are going to do that. 

Regarding inlining submission box or leaving it at the left side, I am not quite about it. I have found sites with both approaches and I am not totally convinced that one of them is superior to the other. 

My guess is that having it on the left side lets you read and code at the same time. This is good for more complex exercises. But perhaps for simpler exercises it is not the best idea. Perhaps we should try it and do some A/B testing.
